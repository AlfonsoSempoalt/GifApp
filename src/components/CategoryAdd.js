import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CategoryAdd = ({ setCategories }) => {
  const [category, setCategory] = useState('');

  const handleOnChangeInput = (e) => {
    setCategory(e.target.value);
  };

  const handleOnSubmit = (e) => {
    e.preventDefault();
    if (category.trim().length > 1) {
      setCategories((categories) => [category, ...categories]);
      setCategory('');
    }
  };

  const handleOnDelete = () => {
    setCategories(['']);
  };

  return (
    <>
      <form onSubmit={handleOnSubmit}>
        <input type='text' value={category} onChange={handleOnChangeInput} />
      </form>
      <input type='submit' value='Limpiar Busqueda' onClick={handleOnDelete} />
    </>
  );
};

CategoryAdd.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default CategoryAdd;
