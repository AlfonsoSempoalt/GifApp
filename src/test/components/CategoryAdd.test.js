import { describe, expect } from '@jest/globals';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import React from 'react';
import CategoryAdd from '../../components/CategoryAdd';

describe('tests to <CategoryAdd />', () => {
  const setCategories = jest.fn();
  //esta inicializaci√≥n esta hecha
  //porque si no la hago se pierde el auto completado
  let wrapper = shallow(<CategoryAdd setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<CategoryAdd setCategories={setCategories} />);
  });

  test('should match with snapshot categoryAdd', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should change the text box', () => {
    const value = 'One piece';
    wrapper.find('#inputText').simulate('change', { target: { value } });
    const inputTextValue = wrapper.find('#inputText').prop('value');
    expect(inputTextValue).toBe(value);
  });

  test('should not submit of the form', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('should call setCategories and clear the text input', () => {
    const value = 'Hola mundo';

    wrapper.find('#inputText').simulate('change', { target: { value: value } });
    wrapper.find('form').simulate('submit', { preventDefault() {} });

    expect(setCategories).toHaveBeenCalledWith( expect.any( Function ) );
    expect(wrapper.find('#inputText').prop('value')).not.toBe(value);
  });
});
