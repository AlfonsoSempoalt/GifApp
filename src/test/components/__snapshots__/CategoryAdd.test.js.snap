// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests to <CategoryAdd /> should match with snapshot categoryAdd 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      id="inputText"
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
  <input
    id="clearBtn"
    onClick={[Function]}
    type="submit"
    value="Limpiar Busqueda"
  />
</Fragment>
`;
